core/id/internal_id_wrappers.go

```diff
@@ -0,0 +1,441 @@
+// Copyright 2025 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package id
+
+import "strconv"
+
+// InternalAccessMethod is an Internal wrapper for access methods. This wrapper must not be returned to the client.
+type InternalAccessMethod Internal
```

Comment:
I think I would get rid of `Internal` from all these type aliases and methods, `id.Check` is pretty clear. Ditto `id.NewCheck()`

Reply:
For that matter, can we just call `Internal` `Id` instead? `Internal` doesn't really tell me anything useful about the type and has confusing connotations.